static/app/views/stories/storyIndex.tsx

```diff
@@ -0,0 +1,188 @@
+import {useLayoutEffect, useMemo, useState} from 'react';
+import styled from '@emotion/styled';
+
+import TextOverflow from 'sentry/components/textOverflow';
+import {space} from 'sentry/styles/space';
+
+type Entry = {
+  ref: HTMLElement;
+  title: string;
+};
+
+function getContentEntries(main: HTMLElement): Entry[] {
+  const titles = main.querySelectorAll('h2, h3, h4, h5, h6');
+  const entries: Entry[] = [];
+
+  for (const entry of Array.from(titles ?? [])) {
+    // Ensure each title has an id we can link to
+    if (!entry.id) {
+      entry.id = entry.textContent?.replace(/ /g, '-').toLowerCase() ?? '';
```

Comment:
Seems fine to roll our own instead of pulling in a dependency like `slugify`, but we might want a more exhaustive check for other non-alphanumeric values here, like `/[^a-z0-9\s-]/g`?

Reply:
Good catch, I changed it to enforce alphanumeric here